src/mbgl/tile/geometry_tile_data.cpp

```diff
@@ -180,4 +180,11 @@ Feature convertFeature(const GeometryTileFeature& geometryTileFeature, const Can
     return feature;
 }
 
+Feature convertFeatureProperties(const GeometryTileFeature& geometryTileFeature) {
+    Feature feature { Point<double>() };
```

Comment:
Forcing all features to be of type `point` means that the results will not be able to tell the `geometry_type` of a feature. 
How about a `switch` statement to create the `feature` with the correct geometry type ?